src/modules/referral/services/service.ts

```diff
@@ -71,4 +75,50 @@ export class ReferralService {
 
     return { referralRate: 0.4, tier: 1 };
   }
+
+  public subtractFunctionArgsFromCallData(data: string) {
+    const coder = new ethers.utils.AbiCoder();
+    // strip hex method identifier
+    const dataNoMethod = ethers.utils.hexDataSlice(data, 4);
+    // keep method hex identifier
+    const methodHex = data.replace(dataNoMethod.replace("0x", ""), "");
+    const decodedData = coder.decode(
```

Comment:
Would this work if the transaction is sent to a non Across contract, like what Socket uses? They use some other contract that calls Across's spoke pool

Reply:
I had the same question. I didn't test it extensively because I saw that all the deposits were getting filled in the database. This means that all the event parameters: deposit id, chain id, amount, etc were decoded correctly.